app/facades/trust_registry.py

```diff
@@ -1,25 +1,120 @@
+
+    Args:
+        did (str): the did of the issuer in fully qualified format.
+        schema_id (str): the schema_id of the credential being issued
+
+    Raises:
+        Exception: When the did is not registered, the actor doesn't have the issuer role
+            or the schema is not registered in the registry.
+    """
+    actor = await actor_by_did(did)
+
+    if not actor:
+        raise TrustRegistryException(f"Did {did} not registered in the trust registry")
+
+    actor_id = actor["id"]
+    if not "issuer" in actor["roles"]:
+        raise TrustRegistryException(
+            f"Actor {actor_id} does not have required role 'issuer'"
+        )
+
+    has_schema = await registry_has_schema(schema_id)
+    if not has_schema:
+        raise TrustRegistryException(
+            f"Schema with id {schema_id} is not registered in trust registry"
+        )
+
+
+async def assert_valid_verifier(did: str, schema_id: str):
+    """Assert that an actor with the specified did is registered as verifier.
+
+    This method asserts that there is an actor registered in the trust registry
+    with the specified did. It verifies whether this actor has the `verifier` role
+    and will also make sure the specified schema_id is regsitred as a valid schema.
+    Raises an exception if one of the assertions fail.
+
+    NOTE: the dids in the registry are registered as fully qualified dids. This means
+    when passing a did to this method it must also be fully qualified (e.g. `did:sov:xxxx`)
+
+    Args:
+        did (str): the did of the verifier in fully qualified format.
+        schema_id (str): the schema_id of the credential being issued
+
+    Raises:
+        Exception: When the did is not registered, the actor doesn't have the verifier role
+            or the schema is not registered in the registry.
+    """
+    actor = await actor_by_did(did)
+
+    if not actor:
+        raise TrustRegistryException(f"Did {did} not registered in the trust registry")
+
+    actor_id = actor["id"]
+    if not "verifier" in actor["roles"]:
+        raise TrustRegistryException(
+            f"Actor {actor_id} does not have required role 'verifier'"
+        )
+
+    has_schema = await registry_has_schema(schema_id)
+    if not has_schema:
+        raise TrustRegistryException(
+            f"Schema with id {schema_id} is not registered in trust registry"
+        )
+
+
 async def actor_has_role(actor_id: str, role: Role) -> bool:
     actor_res = requests.get(TRUST_REGISTRY_URL + f"/registry/actors/{actor_id}")
     if actor_res.status_code != 200:
         raise HTTPException(404, detail="Actor does not exist")
     return bool(role in actor_res.json()["roles"])
 
 
-async def actors_with_role(role: str = Role) -> list:
+async def actor_by_did(did: str) -> Optional[Actor]:
+    actor_res = requests.get(TRUST_REGISTRY_URL + f"/registry/actors/did/{did}")
+
+    if actor_res.status_code != 200:
```

Comment:
This is nice and concise. I'd maybe say though that it's worth it being a bit more precise here. Say for instance the TR return 404 not found, then returning none is fine. There is also a chance that the TR is down or busy or conneciton dropped or whatever.

I'd check what exactly one gets for /registry/actors/did/{did} with did non existent and return none only for that case and raise an exception otherwise. That's simply to avoid confusion when using it and the user/program is not false tricked into thinking there is no actor with that did when there is an actor but sth else wnet wrong